### PR TITLE
A couple of features

### PR DESCRIPTION
**Pull request by slang25**
- Make things a little more async
- Search up directories for `global.json` to be consistent with dotnet cli
- Check already installed SDKs and skip where appropriate

### CODE REVIEW

**Comment by josephwoodward** [comment]
Oh nice one, been meaning to add checks if the SDK is already installed. I'll pull this down locally and test it on Windows before the next release.

File: src/DotNetInstallSdk/SdkAcquirer.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Text.Json;
 using System.Threading.Tasks;
 using DotNet.InstallSdk.Acquirables;
+using static SimpleExec.Command;
```
Comment (josephwoodward) [inline src/DotNetInstallSdk/SdkAcquirer.cs]: Nice!

**Comment by slang25** [comment]
Thanks, I've only tested on macOS as usual

File: src/DotNetInstallSdk/SdkAcquirer.cs [hunk 2]
```diff
@@ -113,5 +120,29 @@ void CheckHash(string filePath, string fileHash)
...
+        {
+            try
+            {
+                var dotnetInfo =
+                    (await ReadAsync("dotnet", "--info", noEcho: true)).Split(new[] {'\r', '\n'},
+                        StringSplitOptions.RemoveEmptyEntries);
+
+                var existingSdks =
+                    dotnetInfo
+                        .SkipWhile(x => !x.Contains(".NET Core SDKs installed:")) // TODO localize?
```
Comment (slang25) [inline src/DotNetInstallSdk/SdkAcquirer.cs]: It's not localized, will remove comment

**Comment by slang25** [comment]
Added silent installs (not tested at all! Yeehaw 🤠)
Need to fix failing test (`_global.json` is found in parent folder).

**Comment by josephwoodward** [comment]
> Added silent installs (not tested at all! Yeehaw )

It'll work fine 😁  Ship it!!

**Comment by slang25** [comment]
Ha. So you have to sudo on macOS, which makes sense. Haven't tried it on Windows, other than that I think it's ready, let me know what you think.

File: src/DotNetInstallSdk/InstallerLauncher.cs [hunk 3]
```diff
@@ -9,15 +11,34 @@ public interface IInstallerLauncher
...
-            Process.Start(processStartInfo)?.WaitForExit();
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                Run("sudo", $"installer -pkg {installerPath} -target /");
+                return;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Run(installerPath, $"/install /quiet /norestart");
```
Comment (josephwoodward) [inline src/DotNetInstallSdk/InstallerLauncher.cs:37]: What are your thoughts around making this a flag, either a flag to use headless installs, or a flag to disable a headless install? For instance:

`$ dotnet install-sdk --headless true`

Comment (josephwoodward) [reply src/DotNetInstallSdk/InstallerLauncher.cs:37]: I'd imagine the majority of the time people would prefer for the install to be headless.

Comment (slang25) [reply src/DotNetInstallSdk/InstallerLauncher.cs:37]: That's a good call, and I agree that the default should probably be silent, even though this changes the current behaviour.

Comment (josephwoodward) [reply src/DotNetInstallSdk/InstallerLauncher.cs:37]: Yeah, we can release it as a new version number. Better to get these kind of changes ironed out early on.

**Pull request merged**